### PR TITLE
New K-Fold interface for Splitter

### PR DESCRIPTION
**Pull request by lilleswing**
``` python
    """
    :param dataset: Dataset to do a k-fold split
    :param k: number of folds
    :param directories: list of length 2*k to save the result disk-datasets
    :param kwargs:
    :return: list of length k tuples of (train, cv) datasets
    """
```

### CODE REVIEW

**Comment by lilleswing** [comment]
https://github.com/deepchem/deepchem/issues/677

File: deepchem/data/datasets.py [hunk 1]
```diff
@@ -395,6 +395,14 @@ def select(self, indices, select_dir=None):
     ids = self.ids[indices]
     return NumpyDataset(X, y, w, ids)
 
+  @staticmethod
+  def from_DiskDataset(ds):
+    """
+    :param ds: DiskDataset
```
Comment (rbharath) [inline deepchem/data/datasets.py]: The comment style here looks different from our usual?

Comment (lilleswing) [inline deepchem/data/datasets.py]: blerg I'll re-update to numpy

File: deepchem/splits/splitters.py [hunk 2]
```diff
@@ -54,30 +54,48 @@ def __init__(self, verbose=False):
     self.verbose = verbose
 
   def k_fold_split(self, dataset, k, directories=None, **kwargs):
-    """Does K-fold split of dataset."""
+    """
+    :param dataset: Dataset to do a k-fold split
```
Comment (rbharath) [inline deepchem/splits/splitters.py]: Would you mind changing this docstring as well?

**Comment by lilleswing** [comment]
@rbharath updated the docstrings

**Comment by rbharath** [comment]
LGTM

**Pull request merged**